extensions/elytron-security-oauth2/runtime/src/main/java/io/quarkus/elytron/security/oauth2/runtime/oauth/Oauth2Credential.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.elytron.security.oauth2.runtime.oauth;
+
+import io.undertow.security.idm.Credential;
+
+public class Oauth2Credential implements Credential {
+    private String bearerToken;
+
+    public Oauth2Credential(String bearerToken) {
+        this.bearerToken = bearerToken;
+    }
+
+    public String getBearerToken() {
+        return bearerToken;
+    }
+
+    public String getName() {
+        //TODO names should be retrieved from the principal
+        return null;
```

Comment:
+1 :-)

Reply:
I add a TODO here because in smallrye-jwt the name is searched in the claims that are retrieved from a context. As I really don't like contextes I didn't implement the same but I'm not sure if we really need it.
In fact, a Credential is not mandatory to have a name as the Principal is used in client code not directly the Credential, so I propose to remove it if it's OK for you (will need some refactoring on the AuthMechanism and the IdentityManager classes)

Reply:
if it is not needed then surely lets remove it

Reply:
removed